vortex-layout/src/layouts/chunked/eval_stats.rs

```diff
@@ -10,9 +12,30 @@ use crate::StatsEvaluator;
 impl StatsEvaluator for ChunkedReader {
     async fn evaluate_stats(
         &self,
-        field_paths: &[FieldPath],
-        _stats: &[Stat],
+        field_paths: Arc<[FieldPath]>,
+        stats: Arc<[Stat]>,
     ) -> VortexResult<Vec<StatsSet>> {
-        Ok(vec![StatsSet::default(); field_paths.len()])
+        if field_paths.is_empty() {
+            return Ok(vec![]);
+        }
+
+        // Otherwise, fetch the stats table
+        let Some(stats_table) = self.stats_table().await? else {
+            return Ok(vec![StatsSet::empty(); field_paths.len()]);
+        };
+
+        let mut stat_sets = Vec::with_capacity(field_paths.len());
+        for field_path in field_paths.iter() {
+            if !field_path.is_root() {
+                // TODO(ngates): the stats table only stores a single array, so we can only answer
+                //  stats if the field path == root.
+                //  See <https://github.com/spiraldb/vortex/issues/1835> for more details.
+                stat_sets.push(StatsSet::empty());
+                continue;
+            }
```

Comment:
You don't seem to handle this case, i.e. you never zip the result back with the requested fieldpaths. I would either throw for now in this case or change this function such that you return more information, i.e. you either pad the returned Vec with empty options, return a map or return vec of tuples

Reply:
Isn't that what `stat_sets.push(StatsSet::empty());` does...?

Reply:
sorry, I completely missed that line.